apps/explorer/src/pages/transaction-result/transaction-summary/ObjectChanges.tsx

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getGroupByOwner, LocationIdType } from '@mysten/core';
+import { ChevronRight12 } from '@mysten/icons';
+import {
+    type SuiObjectChangeCreated,
+    type SuiObjectChangeMutated,
+    type SuiObjectChangeTransferred,
+} from '@mysten/sui.js';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import {
+    ExpandableList,
+    ExpandableListControl,
+    ExpandableListItems,
+} from '~/ui/ExpandableList';
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { Text } from '~/ui/Text';
+import {
+    TransactionBlockCard,
+    TransactionBlockCardSection,
+} from '~/ui/TransactionBlockCard';
+
+enum Labels {
+    created = 'Created',
+    mutated = 'Updated',
+    transferred = 'Transfer',
+}
+
+enum ItemLabels {
+    package = 'Package',
+    module = 'Module',
+    function = 'Function',
+}
+
+const DEFAULT_ITEMS_TO_SHOW = 5;
+
+interface ObjectChangeEntryBaseProps {
+    type: keyof typeof Labels;
+}
+
+function Item({
+    label,
+    packageId,
+    moduleName,
+    functionName,
+}: {
+    label: ItemLabels;
+    packageId: string;
+    moduleName: string;
+    functionName: string;
+}) {
+    return (
+        <div className="flex justify-between gap-10">
+            <Text variant="pBody/medium" color="steel-dark">
+                {label}
+            </Text>
+
+            {label === ItemLabels.package && (
+                <ObjectLink objectId={packageId} />
+            )}
+            {label === ItemLabels.module && (
+                <ObjectLink
+                    objectId={`${packageId}?module=${moduleName}`}
+                    label={moduleName}
+                />
+            )}
+            {label === ItemLabels.function && (
+                <Text truncate variant="pBody/medium" color="steel-darker">
+                    {functionName}
+                </Text>
+            )}
+        </div>
+    );
+}
+
+function ObjectDetail({
+    objectType,
+    objectId,
+}: {
+    objectType: string;
+    objectId: string;
+}) {
+    const [expanded, setExpanded] = useState(false);
+    const toggleExpand = () => setExpanded((prev) => !prev);
+
+    const [packageId, moduleName, functionName] =
```

Comment:
is this always consistent or prone to breaking? it's kind of hard to tell what's going on with the double split

Reply:
I am taking this implementation from `wallet` mostly. I assume if that's working there, it should be okay for me